src/main/java/at/hannibal2/skyhanni/features/garden/farming/GardenCustomKeybinds.kt

```diff
@@ -67,43 +76,96 @@ object GardenCustomKeybinds {
         lastDuplicateKeybindsWarnTime = SimpleTimeMark.now()
     }
 
+    @HandleEvent
+    fun onGardenToolChange(event: GardenToolChangeEvent) {
+        lastToolSwitch = SimpleTimeMark.now()
+        cropInHand = event.crop
+        event.crop?.let { lastCrop = it }
+        currentLayout = layouts[cropLayoutSelection[cropInHand]]
+    }
+
     @HandleEvent
     fun onConfigLoad(event: ConfigLoadEvent) {
-        with(config) {
-            ConditionalUtils.onToggle(attack, useItem, left, right, forward, back, jump, sneak) {
-                update()
-            }
+        var allKeybindings = listOf(
+            config.layout1.attack, config.layout1.useItem, config.layout1.left, config.layout1.right,
+            config.layout1.forward, config.layout1.back, config.layout1.jump, config.layout1.sneak,
+            config.layout2.attack, config.layout2.useItem, config.layout2.left, config.layout2.right,
+            config.layout2.forward, config.layout2.back, config.layout2.jump, config.layout2.sneak,
+            config.layout3.attack, config.layout3.useItem, config.layout3.left, config.layout3.right,
+            config.layout3.forward, config.layout3.back, config.layout3.jump, config.layout3.sneak,
+            config.layout4.attack, config.layout4.useItem, config.layout4.left, config.layout4.right,
+            config.layout4.forward, config.layout4.back, config.layout4.jump, config.layout4.sneak,
+            config.layout5.attack, config.layout5.useItem, config.layout5.left, config.layout5.right,
+            config.layout5.forward, config.layout5.back, config.layout5.jump, config.layout5.sneak
```

Comment:
Small Suggestion:
Better to add a function that gets all properties as a list and then simply combine the 5 lists. Makes it better scale able.

Reply:
@Thunderblade73 
would it be better to use stuff like getting all the variables in the config to then dynamically do the stuff. so you could just adjust the number of layouts in config and then the kotlin would automatically just work with the amount of layouts in the config?
My main concern with this is the unneeded complexity this would generate

Reply:
@Thunderblade73 
like the new version?